.github/workflows/ci.yaml

```diff
@@ -183,3 +182,98 @@ jobs:
           emulator-options: -no-snapshot-save -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
           disable-animations: true
           script: ./gradlew connectedCheck -Pandroid.testInstrumentationRunnerArguments.class=org.ZingoLabs.Zingo.IntegrationTestSuite
+  
+  e2e_test:
+    name: End-to-End Test
+    runs-on: macos-12
```

Comment:
I think the best person to review this is @Oscar-Pepper, but from my ignorance I saw `macos-12` here and I'm confused because the rest of the code seems to be for Android...

Reply:
I will review first thing when I'm back! Yes so we usually use the Ubuntu runners for most jobs but for emulation (even android emulation!) MacOS is better as it has hardware acceleration etc.